### ISSUE TITLE
ModalBarrier behind navigation drawer is missing a label

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#19899]
On iOS, open the Material Search demo in the gallery. Open the navigation drawer and focus with VoiceOver on the barrier. It should read out a label, e.g. "close", but it doesn't.
